Connection timeouts since version 2.6.7 or 2.6.6

### Describe the bug

We upgraded the aws-advanced-jdbc-wrapper from version 2.6.5 to 2.6.7. Since then, we have the problem that a connection times out when the Aurora Serverless database we are using, is not up. This was not a problem before. We are using the following parameters for connections:

- `wrapperPlugins=initialConnection`
- `openConnectionRetryTimeoutMs=180000`
- `loginTimeout=40000`
- `connectTimeout=40000`

Before, we had lower values for the timeouts but increased them, because we assumed that this would fix the problem.

### Expected Behavior

If the Aurora Serverless database is not up, we expect that the connection attempt is not immediately aborted but a retry happens for the timeout specified.

### What plugins are used? What other connection properties were set?

wrapperPlugins=initialConnection&openConnectionRetryTimeoutMs=180000&loginTimeout=40000&connectTimeout=40000

### Current Behavior

It seems like that the connection runs quickly into a timeout.

### Reproduction Steps

We are using that in the Maven DependencyCheck plugin. 

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### The AWS Advanced JDBC Wrapper version used

2.6.7

### JDK version used

OpenJDK 64-Bit Server VM Temurin-17.0.17+10 (build 17.0.17+10, mixed mode, sharing)

### Operating System and version

Ubuntu 24.04.3 LTS

I tested the behaviour again with the new version 2.6.8 and this time, I get a stack trace, when trying to connect to a sleeping Aurora Serverless instance:

```
Exception in thread "main" org.postgresql.util.PSQLException: The connection attempt failed.
	at org.postgresql.Driver$ConnectThread.getResult(Driver.java:397)
	at org.postgresql.Driver.connect(Driver.java:305)
	at software.amazon.jdbc.DriverConnectionProvider.connect(DriverConnectionProvider.java:138)
	at software.amazon.jdbc.plugin.DefaultConnectionPlugin.connectInternal(DefaultConnectionPlugin.java:195)
	at software.amazon.jdbc.plugin.DefaultConnectionPlugin.connect(DefaultConnectionPlugin.java:178)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$connect$6(ConnectionPluginManager.java:350)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$null$0(ConnectionPluginManager.java:237)
	at software.amazon.jdbc.ConnectionPluginManager.executeWithTelemetry(ConnectionPluginManager.java:209)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$makePluginChainFunc$1(ConnectionPluginManager.java:237)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$null$2(ConnectionPluginManager.java:247)
	at software.amazon.jdbc.plugin.AuroraInitialConnectionStrategyPlugin.getVerifiedWriterConnection(AuroraInitialConnectionStrategyPlugin.java:196)
	at software.amazon.jdbc.plugin.AuroraInitialConnectionStrategyPlugin.connect(AuroraInitialConnectionStrategyPlugin.java:148)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$connect$6(ConnectionPluginManager.java:350)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$null$3(ConnectionPluginManager.java:245)
	at software.amazon.jdbc.ConnectionPluginManager.executeWithTelemetry(ConnectionPluginManager.java:209)
	at software.amazon.jdbc.ConnectionPluginManager.lambda$makePluginChainFunc$4(ConnectionPluginManager.java:244)
	at software.amazon.jdbc.ConnectionPluginManager.executeWithSubscribedPlugins(ConnectionPluginManager.java:194)
	at software.amazon.jdbc.ConnectionPluginManager.connect(ConnectionPluginManager.java:347)
	at software.amazon.jdbc.wrapper.ConnectionWrapper.init(ConnectionWrapper.java:111)
	at software.amazon.jdbc.wrapper.ConnectionWrapper.<init>(ConnectionWrapper.java:81)
	at software.amazon.jdbc.Driver.connect(Driver.java:260)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:681)
	at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:229)
	at Test.main(Test.java:17)
Caused by: java.net.SocketTimeoutException: Read timed out
	at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:288)
	at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:314)
	at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:355)
	at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:808)
	at java.base/java.net.Socket$SocketInputStream.read(Socket.java:966)
	at org.postgresql.core.VisibleBufferedInputStream.readMore(VisibleBufferedInputStream.java:162)
	at org.postgresql.core.VisibleBufferedInputStream.ensureBytes(VisibleBufferedInputStream.java:129)
	at org.postgresql.core.VisibleBufferedInputStream.ensureBytes(VisibleBufferedInputStream.java:114)
	at org.postgresql.core.VisibleBufferedInputStream.read(VisibleBufferedInputStream.java:74)
	at org.postgresql.core.PGStream.receiveChar(PGStream.java:467)
	at org.postgresql.core.v3.ConnectionFactoryImpl.enableSSL(ConnectionFactoryImpl.java:594)
	at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:195)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:262)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:54)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:273)
	at org.postgresql.Driver.makeConnection(Driver.java:446)
	at org.postgresql.Driver.access$100(Driver.java:63)
	at org.postgresql.Driver$ConnectThread.run(Driver.java:355)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

When the database is running, the connection works without any problems. When it sleeps and I am using version 2.6.6 it also works without any problem.

--- a/wrapper/src/main/java/software/amazon/jdbc/plugin/AuroraInitialConnectionStrategyPlugin.java
+++ b/wrapper/src/main/java/software/amazon/jdbc/plugin/AuroraInitialConnectionStrategyPlugin.java
@@ -223,6 +223,13 @@ public class AuroraInitialConnectionStrategyPlugin extends AbstractConnectionPlu
           if (writerCandidate != null) {
             this.pluginService.setAvailability(writerCandidate.asAliases(), HostAvailability.NOT_AVAILABLE);
           }
+
+          if (this.pluginService.isNetworkException(ex, this.pluginService.getTargetDriverDialect())
+              || this.pluginService.isReadOnlyConnectionException(ex, this.pluginService.getTargetDriverDialect())) {
+            // Retry connection.
+            continue;
+          }
+
           throw ex;
         }
       } catch (Throwable ex) {

